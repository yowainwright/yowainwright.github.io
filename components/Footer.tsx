import React from 'react'
import Link from 'next/link'

import { SocialFooter } from './SocialFooter'

export const Footer = () => (
  <footer className='site-footer' role='contentinfo' itemType='http://schema.org/WPFooter'>
    <section className='site-footer__wrapper site-footer__wrapper--main'>
      <article className='site-footer__col site-footer__col--contact'>
        <h3 className='site-footer__title'>Contact</h3>
        <address className='site-footer__address'>
          <p className='site-footer__content'>Happy to chat, learn, help!</p>
          <p className='site-footer__content'>
            <a href='mailto:yowainwright@gmail.com'>yowainwright@gmail.com</a>, Los Angeles, CA
          </p>
        </address>
      </article>
      <div className='site-footer__col site-footer__col site-footer__col--social'>
        <h3 className='site-footer__title'>Connect</h3>
        <SocialFooter />
      </div>
      <article className='site-footer__col site-footer__col--self'>
        <h3 className='site-footer__title site-footer__title--self'>About</h3>
        <figure className='site-footer__figure'>
          <img
            className='media--circular site-footer__image'
            src='https://yowainwright.imgix.net/portraits/me-smiling.jpg?w=150&h=150&fit=crop&auto=format'
            height='100%'
            width='100%'
            alt='Me smiling with a beard'
          />
          <figcaption className='site-footer__caption'>
            <Link className='site-footer__link' href='/'>
              <strong>jeffry.in</strong>, {new Date().getFullYear()}
            </Link>
            , is the blog of{' '}
            <Link className='site-footer__link' href='/about'>
              Jeffry (Jeff) Wainwright
            </Link>
            , a software engineer, person, living in Los Angeles.
          </figcaption>
        </figure>
      </article>
    </section>
    <section className='site-footer__wrapper site-footer__wrapper--last'>
      <h3 className='site-footer__title site-footer__title--last'>
        <Link className='site-footer__link' href='/'>
          jeffry.in
        </Link>
        , {new Date().getFullYear()}
      </h3>
    </section>
  </footer>
)

export default Footer
