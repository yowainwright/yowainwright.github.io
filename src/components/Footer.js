import React from "react"

class Footer extends React.Component {
  render() {
    return (
      ~
      {" "}
      <footer
        class="site-footer"
        role="contentinfo"
        itemtype="http://schema.org/WPFooter"
      >
        ~
        {" "}
        <p class="site-footer__content">
          <Link
            class="site-footer__link"
            to={'/''>
              jeffry.in
          </LInk>
          ~
          {" "}
          <span
            class="site-footer__date"
            data-date="site-year">
              2017
          </span>
        </p>
      </footer>
    )
  }
}

export default Footer
