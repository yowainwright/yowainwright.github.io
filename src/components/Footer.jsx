import React from 'react'
import { Link } from 'gatsby'

import { SocialFooter } from './SocialFooter';

export const Footer = () => (
  <footer className='site-footer' role='contentinfo' itemType='http://schema.org/WPFooter'>
    <section className='site-footer__wrapper site-footer__wrapper--main'>
      <article className='site-footer__col site-footer__col--self'>
        <h3 className='site-footer__title site-footer__title--self'>
          <Link className='site-footer__link' to='/about'>
            About
          </Link>
        </h3>
        <figure className='site-footer__figure'>
          <img
            className='media--circular site-footer__image'
            src='https://yowainwright.imgix.net/portraits/me-smiling.jpg?w=100&h=100&fit=crop&auto=format'
            alt='Me smiling with a beard'
          />
          <figcaption className='site-footer__caption'>
            <Link className='site-footer__link' to='/'>
              <strong>jeffry.in</strong>, {new Date().getFullYear()}
            </Link>, is the blog of Jeffry (Jeff) Wainwright, a software engineer, person, living in Los Angeles.
          </figcaption>
        </figure>
      </article>
      <div className='site-footer__col site-footer__col site-footer__col--social'>
        <h3 className='site-footer__title'><a href='https://github.com/yowainwright'>Connect</a></h3>
        <SocialFooter />
      </div>
     <article className='site-footer__col site-footer__col--contact'>
        <h3 className='site-footer__title'>
          <a href='mailto:yowainwright@gmail.com'>Contact</a>
        </h3>
        <address className='site-footer__address'>
          <ul className='site-footer__items site-footer__items--address'>
            <li className='site-footer__item'>
              Happy to chat, learn, help! <span className='site-footer__thumbs'></span>
            </li>
            <li className='site-footer__item'>
              <a href='mailto:yowainwright@gmail.com'>yowainwright@gmail.com</a>
            </li>
            <li className='site-footer__item'>Los Angeles, CA 90035</li>
        </ul>
      </address>
      </article>
    </section>
    <h3 className='site-footer__title site-footer__title--last'>
      <Link className='site-footer__link' to='/'>
        jeffry.in, {new Date().getFullYear()} <span className='site-footer__symbol site-footer__symbol--lightning'></span>
      </Link>
    </h3>
  </footer>
)

export default Footer
