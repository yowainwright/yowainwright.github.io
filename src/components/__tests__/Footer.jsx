import React from 'react'
import Footer from '../Footer'

describe('Footer component', () => {
  it('renders correctly', () => {
    const rendered = shallow(<Footer />)

    expect(rendered).toMatchSnapshot()
  })
})
