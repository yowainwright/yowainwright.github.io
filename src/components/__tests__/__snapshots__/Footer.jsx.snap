// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<footer
  className="site-footer"
  itemType="http://schema.org/WPFooter"
  role="contentinfo"
>
  <div
    className="site-footer__wrapper"
  >
    <div
      className="site-footer__col"
    >
      <p
        className="site-footer__body"
      >
        My online thoughts. I'm 
        <Link
          to="/about"
        >
          <strong>
            Jeffry Wainwright
          </strong>
        </Link>
        â€”a software engineer, artist, runner, and surfer living in Los Angeles. 
        <strong>
          <a
            href="mailto:yowainwright@gmail.com"
          >
            Email me
          </a>
        </strong>
         or reach out on 
        <a
          href="https://github.com/yowainwright"
        >
          <strong>
            Github
          </strong>
        </a>
        , 
        <a
          href="https://instagram.com/yowainwright"
        >
          <strong>
            Instagram
          </strong>
        </a>
        , 
        <a
          href="https://twitter.com/yowainwright"
        >
          <strong>
            Twitter
          </strong>
        </a>
        , 
        <a
          href="https://www.strava.com/athletes/722335"
        >
          <strong>
            Strava
          </strong>
        </a>
        , or 
        <a
          href="https://www.linkedin.com/in/jeffrywainwright/"
        >
          <strong>
            LinkedIn
          </strong>
        </a>
        .
      </p>
    </div>
    <div
      className="site-footer__col"
    >
      <nav
        className="social-footer"
      >
        <ul
          className="social-list"
        >
          <li
            className="social-list__item social-list__item--showing"
          >
            <a
              className="social-list__link"
              href="https://github.com/yowainwright"
            >
              Github
            </a>
          </li>
          <li
            className="social-list__item social-list__item--showing"
          >
            <a
              className="social-list__link"
              href="https://twitter.com/yowainwright"
            >
              Twitter
            </a>
          </li>
          <li
            className="social-list__item social-list__item--hidden"
          >
            <a
              className="social-list__link"
              href="https://www.npmjs.com/~yowainwright"
            >
              NPMJS
            </a>
          </li>
          <li
            className="social-list__item social-list__item--showing"
          >
            <a
              className="social-list__link"
              href="https://instagram.com/yowainwright"
            >
              Instragram
            </a>
          </li>
          <li
            className="social-list__item social-list__item--showing"
          >
            <a
              className="social-list__link"
              href="https://www.linkedin.com/in/jeffrywainwright/"
            >
              LinkedIn
            </a>
          </li>
          <li
            className="social-list__item social-list__item--showing"
          >
            <a
              className="social-list__link"
              href="/rss.xml"
            >
              RSS
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="site-footer__col"
    >
      <figure
        className="figure--center"
      >
        <img
          alt="Me smiling with a beard"
          className="media--circular"
          src="https://yowainwright.imgix.net/portraits/me-smiling.jpg?w=100&h=100&fit=crop&auto=format"
        />
      </figure>
      <p
        className="site-footer__content"
      >
        <Link
          className="site-footer__link"
          to="/"
        >
          jeffry.in 
          2020
        </Link>
      </p>
    </div>
  </div>
</footer>
`;
